### PR TITLE
Various fixes for test_sync_safekeepers

### PR DESCRIPTION
**Pull request by petuhovskiy**
- Fixed flaky behavior by sending ProposerGreeting manually
- Test moved to test_wal_acceptor.py
- Test output is now captured to files

You can look at 1st and 3rd commit to see what's really changed.

### CODE REVIEW

File: test_runner/batch_others/test_wal_acceptor.py [hunk 1]
```diff
@@ -198,3 +202,90 @@ def test_race_conditions(zenith_cli, pageserver: ZenithPageserver, postgres: Pos
...
+
+    def create_dir_config(self, wal_acceptors: str):
+        """ Create dir and config for running --sync-safekeepers """
+
+        mkdir_if_needed(self.pg_data_dir_path())
+        with open(self.config_file_path(), "w") as f:
+            f.write("zenith.zenith_timeline = '{}'\n".format(self.timeline_id))
+            f.write("zenith.zenith_tenant = '{}'\n".format(self.tenant_id))
+            f.write("synchronous_standby_names = '{}'\n".format("walproposer"))
+            f.write("wal_acceptors = '{}'\n".format(wal_acceptors))
```
Comment (funbringer) [inline test_runner/batch_others/test_wal_acceptor.py:230]: IMO f-strings would look great here.

**Review by funbringer — COMMENTED** [review]
Just my superficial nits

**Comment by petuhovskiy** [comment]
Strange error occurred while running tests locally:

```

self = WalAcceptor(wa_bin_path=PosixPath('/Users/arthur/zen/zenith/target/debug/wal_acceptor'), data_dir=PosixPath('/Users/ar...t/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_0'), port=15002, num=0, pageserver_port=15000, auth_token=None)
tenant_id = '4afa5cf430794e328da1fdd9bcdd9e61', timeline_id = 'ea49304435094672baf2db38c4e9ad24', request = {'begin_lsn': 23826824, 'epoch_start_lsn': 23826824, 'lm_message': 'message', 'lm_prefix': 'prefix', ...}

    def append_logical_message(self, tenant_id: str, timeline_id: str, request: Dict[str, Any]) -> Dict[str, Any]:
        """
        Send JSON_CTRL query to append LogicalMessage to WAL and modify
        safekeeper state. It will construct LogicalMessage from provided
        prefix and message, and then will write it to WAL.
        """

        # "replication=0" hacks psycopg not to send additional queries
        # on startup, see https://github.com/psycopg/psycopg2/pull/482
        connstr = f"host=localhost port={self.port} replication=0 options='-c ztimelineid={timeline_id} ztenantid={tenant_id}'"

        with closing(psycopg2.connect(connstr)) as conn:
            # server doesn't support transactions
            conn.autocommit = True
            with conn.cursor() as cur:
                request_json = json.dumps(request)
                print(f"JSON_CTRL request on port {self.port}: {request_json}")
>               cur.execute("JSON_CTRL " + request_json)
E               psycopg2.DatabaseError: failed to open control file "/Users/arthur/zen/zenith/test_output/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_0/ea49304435094672baf2db38c4e9ad24/safekeeper.control": No such file or directory (os error 2)

fixtures/zenith_fixtures.py:877: DatabaseError
--------------------------------------------------------------------------------------------------------- Captured stdout setup ----------------------------------------------------------------------------------------------------------

base_dir is /Users/arthur/zen/zenith
test_output_dir is /Users/arthur/zen/zenith/test_output/test_sync_safekeepers
postgres dir is /Users/arthur/zen/zenith/tmp_install
pageserver_port: pg=15000 http=15001
---------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------
Running command "/Users/arthur/zen/zenith/target/debug/wal_acceptor -D /Users/arthur/zen/zenith/test_output/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_0 -l localhost:15002 --daemonize --no-sync --pageserver localhost:15000 --recall 1 second"
Running command "/Users/arthur/zen/zenith/target/debug/wal_acceptor -D /Users/arthur/zen/zenith/test_output/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_1 -l localhost:15003 --daemonize --no-sync --pageserver localhost:15000 --recall 1 second"
Running command "/Users/arthur/zen/zenith/target/debug/wal_acceptor -D /Users/arthur/zen/zenith/test_output/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_2 -l localhost:15004 --daemonize --no-sync --pageserver localhost:15000 --recall 1 second"
JSON_CTRL request on port 15002: {"lm_prefix": "prefix", "lm_message": "message", "set_commit_lsn": true, "term": 2, "begin_lsn": 23826824, "epoch_start_lsn": 23826824, "truncate_lsn": 23826824}
-------------------------------------------------------------------------------------------------------- Captured stdout teardown --------------------------------------------------------------------------------------------------------
Starting wal acceptors cleanup
Stopping wal acceptor 0
Stopping wal acceptor 1
Stopping wal acceptor 2
======================================================================================================== short test summary info =========================================================================================================
FAILED batch_others/test_wal_acceptor.py::test_sync_safekeepers - psycopg2.DatabaseError: failed to open control file "/Users/arthur/zen/zenith/test_output/test_sync_safekeepers/repo/wal_acceptors/wal_acceptor_0/ea49304435094672baf...

```

**Comment by arssher** [comment]
> Strange error occurred while running tests locally:

Hm. That's even after you added `query_string.starts_with(b"JSON_CTRL"` to send_wal.rs::56, right?

**Comment by arssher** [comment]
Let's add a comment to `handle_json_ctrl`, explaining it requests to craft logical message WAL record with given contents and append it with given term, which is used for tests.

**Review by arssher — APPROVED** [review]

**Pull request merged**